packages/loader/driver-utils/src/network.ts

```diff
@@ -112,8 +112,16 @@ export class AuthorizationError
 	implements IAuthorizationError, IFluidErrorBase
 {
 	readonly errorType = DriverErrorTypes.authorizationError;
-	readonly claims?: string;
-	readonly tenantId?: string;
+
+	// These properties are not assigned in this class, but instead assigned in the super constructor.
+	// When targeting ES 2022 or later, TypeScript generates ES6 class fields for these properties.
+	// This overrides the own properties dynamically created by the super constructor.
```

Comment:
typo - perhaps you mean to omit "own"

But might warrant some rewording. I think you mean to say that `readonly claims?: string` would override super class properties. It is hard to understand because *now* it says `declare ...` which does not have that behvior.

Maybe something like:
When targeting ES 2022 or later, TypeScript generates ES6 class fields for properties listed without `declare` keyword and overrides the properties dynamically created by the super constructor.

Or just leave it out and amend the following to say:
To prevent undesired overriding of super class properties (when targeting ES 2022 or later),

Reply:
I specifically mean "own properties" (the same "own properties" in APIs like "Object.getOwnPropertyNames").

I specifically mentioned this, since overriding inherited properties is much more common and easier, but not what is being done here.

How about:
```suggestion
	// When targeting ES 2022 or later, TypeScript would generate ES6 class fields for these properties if they did not use "declare".
	// That would override the own properties dynamically created in the super constructor
	// resulting in these properties always holding `undefined` instead of their desired values.
```

Reply:
I think that is more clear: Does that address your concerns?